lsat/interceptor_test.go

```diff
@@ -240,6 +240,45 @@ func TestInterceptor(t *testing.T) {
 		return backendErr
 	}
 
+	// We'll duplicate all test cases and run through the first half with
+	// the unary interceptor and then through the second half with the
+	// stream interceptor.
+	for i := 0; i < len(testCases); i++ {
+		tc := testCases[i]
+		unaryInvoker := func(_ context.Context, _ string, _ interface{},
+			_ interface{}, _ *grpc.ClientConn,
+			opts ...grpc.CallOption) error {
+
+			return invoker(opts)
+		}
+
+		testCases[i].callInterceptor = func() error {
+			return tc.interceptor.UnaryInterceptor(
+				ctx, "", nil, nil, nil, unaryInvoker, nil,
+			)
+		}
+	}
+
+	// Duplicate the test cases now and set the streaming interceptor for
+	// the second set.
+	doubleTestCases := append(testCases, testCases...)
```

Comment:
I don't think the stream tests are being run in this diff, don't we need to range over them below?

Also, by taking this approach, we'll have duplicate names for the unary and stream interceptors. I think a cleaner approach would be to refactor all common operations within the tests into helpers and creating a separate top level test for the stream interceptors, such that we have `TestUnaryInterceptor` and `TestStreamInterceptor`. I wouldn't consider this a blocker though.

Reply:
Haha, yes. I thought it was too easy that the tests just worked.
I split the two tests to avoid something like that in the future.